src/soundsourceproxy.cpp

```diff
@@ -248,51 +193,53 @@ void SoundSourceProxy::loadPlugins() {
         }
     }
 
-    s_soundSourceProviders.finishRegistration();
-
     const QStringList supportedFileExtensions(
-            s_soundSourceProviders.getSupportedFileExtensions());
-    foreach (const QString &supportedFileExtension, supportedFileExtensions) {
-        const Mixxx::SoundSourceProviderPointer pProvider(
-                s_soundSourceProviders.getProviderForFileExtension(supportedFileExtension));
-        const Mixxx::SoundSourcePluginLibraryPointer pPluginLibrary(
-                s_soundSourceProviders.getPluginLibraryForFileExtension(supportedFileExtension));
-        if (pPluginLibrary) {
-            qDebug() << "SoundSourceProvider for" << supportedFileExtension
-                    << "is" << pProvider->getName()
-                    << "@" << pPluginLibrary->getFilePath();
-        } else {
-            qDebug() << "SoundSourceProvider for" << supportedFileExtension
-                    << "is" << pProvider->getName();
+            s_soundSourceProviders.getRegisteredFileExtensions());
+    for (const auto &supportedFileExtension: supportedFileExtensions) {
+        qDebug() << "SoundSource providers for file extension" << supportedFileExtension;
+        const auto registrationsForFileExtension(
+                s_soundSourceProviders.getRegistrationsForFileExtension(
+                        supportedFileExtension));
```

Comment:
The main purpose of the typedef is not to save any characters ;) I will remove it improves readability.